### PR TITLE
Added publiclab editor,fixes #904

### LINKED ISSUE #904: New form design for asking questions
Currently we re-use the basic note posting form at https://publiclab.org/post for questions. We should revise this to be more specific to, and helpful for, question asking.

Questions are often asked by people new to Public Lab, who will need extra clarity and context. Here's a design which addresses this:

screen shot 2016-10-18 at 12 44 10 pm

To implement, we'll need a new form at https://github.com/publiclab/plots2/blob/master/app/views/editor/ -- maybe called `https://github.com/publiclab/plots2/blob/master/app/views/editor/question.html.erb`?

That would be accessible (i believe) at `/editor/question`. I've created a separate issue for the routing portion to actually use the form: #910. (This is now complete!)

Now we need to tackle integration of the rich editor, at https://github.com/publiclab/PublicLab.Editor/

This can be done similarly to how it was done here: https://github.com/publiclab/plots2/tree/master/app/views/editor/rich.html.erb -- 

Happy to help with this one -- it's big!

### PR DESCRIPTION
**Pull request by 500swapnil**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [ ] all tests pass -- `rake test:all`
* [ ] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [ ] pull request is descriptively named with #number reference back to original issue
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [ ] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by 500swapnil** [comment]
Do I need to integrate the editor only in the "Share your work" textarea?

**Comment by jywarren** [comment]
Hmm, can you elaborate on your last question? 

Also, does this seem to work on your local setup, and could you upload a screenshot so I could take a look? 

Thanks!

**Comment by 500swapnil** [comment]
@jywarren I meant, where should I integrate the publiclab editor? In the "Share your work" text area?
screenshot from 2017-01-12 15-54-03

**Comment by jywarren** [comment]
Ah, i see -- and apologies for the slow response, we've had some outages and also Github was even briefly down (!). 

Actually the code from "rich.html.erb" can replace the entire `<form>` from the questions page. Then we can begin adding/removing things to make it look more like the mockup. Does that make sense?

**Comment by 500swapnil** [comment]
There is no '<form>' tag in the questions page. What do I replace exactly?

**Comment by jywarren** [comment]
Oh, i'm so sorry I missed this and didn't respond earlier! I've been a bit extra busy this past week. Looking now.

File: app/views/editor/questionRich.html.erb [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    <button class="ple-publish btn btn-lg btn-primary pull-right disabled">Publish</button>
+    <span class="ple-help pull-right">
+      <span class="ple-steps-left">2</span>
+      <span class="hidden-xs"> steps left</span>
+    </span>
+    
+  </div>
+
+</div>
+  <%= javascript_include_tag "post" %>
```
Comment (jywarren) [inline app/views/editor/questionRich.html.erb]: See how here, the new "rich" template ends and the older editor template begins? You actually only need one of the two -- that's why you're seeing the old form sort of overlaid on the new one. You should be able to remove everything below this line, but transfer some of the special question-related features from the old template to the new. I'll mark the ones I see.

File: app/views/editor/questionRich.html.erb [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+<div class="col-sm-9">
+  <%
+  if @node && params[:action] != "create"
+    url = { :controller => "notes", :action => "update", :id => params[:id] }
+  else
+    url = { :controller => "notes", :action => "create", :id => params[:id] }
+  end
+  %>
+  <%= form_for @node, :as => :drupal_node, :url => url do |f| %>
```
Comment (jywarren) [inline app/views/editor/questionRich.html.erb]: Here's where the second form is generated. But actually upon reading through it, I don't think there's anything in particular you need to save from this older form. I think you can just delete it and get this template (and its corresponding controller) to match `rich.html.erb` as closely as you can. 

Then we can take the next steps on the design and getting it to look like the mockup.

**Comment by jywarren** [comment]
Hi, just checking if you've gotten stuck on this at all, or if I could help in any way? Thanks!

**Comment by 500swapnil** [comment]
@jywarren Hi, I'm having exams right now. I will continue working on it in a couple of days.

**Comment by jywarren** [comment]
Good luck!

**Comment by 500swapnil** [comment]
Hey @jywarren My exams are over. I will continue working on this issue. I have created another PR #1286

**Comment by jywarren** [comment]
Awesome, thanks!

**Pull request closed without merging**